ODIN_II/SRC/simulate_blif.cpp

```diff
@@ -2222,10 +2222,12 @@ static int verify_test_vector_headers(FILE* in, lines_t* l) {
         error_message(SIMULATION, 0, -1, "%s\n", "Failed to read vector headers.");
 
     // Parse the header, checking each entity against the corresponding line.
-    char buffer[BUFFER_MAX_SIZE];
+    char buffer[BUFFER_MAX_SIZE]; //char array
     buffer[0] = '\0';
+    trim_string(read_buffer," \t\r\n");    //trims string to get rid of unwanted spaces at the end
     unsigned int i;
-    for (i = 0; i < strlen(read_buffer) && i < BUFFER_MAX_SIZE; i++) {
+    for (i = 0; i < strlen(read_buffer) && i < BUFFER_MAX_SIZE; i++) //issue when buffer string has spaces following it following it
+    {
```

Comment:
The fix I want to make is use the trim_string defined in odin_util that trims unwanted characters from a char* string

Reply:
Idk how to comment the line i want you to view because it's 1000 lines away but here 
`/*
 * Gets the next line from the given file that
 * passes the is_vector() test and places it in
 * the buffer. Returns true if a vector was found,
 * and false if no vector was found.
 */
static int get_next_vector(FILE* file, char* buffer) {
    oassert(file != NULL
            && "unable to retrieve file for next test vector");
    oassert(buffer != NULL
            && "unable to use buffer for next test vector as it is not initialized");

    while (fgets(buffer, BUFFER_MAX_SIZE, file))
        if (is_vector(buffer))
            return true;

    return false;
}`

Reply:
it takes a char* buffer pointed but we see earlier that it's an array that's passed hence my confusion

Reply:
a char[] is a char*... somewhat
the first item in a char[] is also a char*.

```C
char buffer[256];
char *ptr = buffer;
// equivalent to
char *ptr = &buffer[0];
// &buffer[0] == buffer == ptr

Reply:
Okay so when it passes the array it really just passes the pointer to the first element of the array?

Reply:
that is correct, it is just handled as a pointer and then you handle both just as if they were pointers or arrays, however you feel to do it

`ptr[1]` would give the value 
`*(ptr + 1)` gives the value
`ptr+1` gives the address, 
`&ptr[1]` gives the address.

they all work with pointer or arrays, doesnt matter, its just different default behavior.

a pointer always gives the address unless you dereference
an array access always gives the value unless you ask for the reference.

so to maybe complicate but drive the point

```C
int a = 31;
int *ptr = &a;

*ptr == 31
ptr[0] == 31
```
it really doesnt matter what you pass as a pointer, its just a memory location and accessing just offsets by the index X the sizeof(type), in this perticular one, it would be 4 byte, wether you are doing pointer arithmetic or array indices